crates/pico/src/memo.rs

```diff
@@ -14,66 +14,104 @@ pub enum DidRecalculate {
     Recalculated,
 }
 
+/// Memo is the workhorse function of pico. Given a [`DerivedNodeId`], which
+/// uniquely identifies the function being called along with the parameters
+/// passed, it:
+///
+/// - checks if we have an available [`DerivedNode`] (i.e. if this function
+///   has previously been invoked, and the [`DerivedNode`] has not been
+///   garbage collected).
+///   - if so, check if the value from the previous invocation can be reused.
+///     - if so, reuse the value from the previous invocation
+///     - if not, re-invoke the function. If the value is the same, backdate
+///       the `DerivedNode`.
+///   - if not, execute this function. Create and store a [`DerivedNode`].
+///
+/// In all cases, we  and set the [`DerivedNode`]'s `verified_at` to the
```

Comment:
In all cases, we set the [`DerivedNode`]'s probably

Reply:
whoops thanks